controllers/secret_controller_test.go

```diff
@@ -0,0 +1,154 @@
+package controllers
+
+import (
+	"testing"
+
+	"github.com/3scale-labs/authorino/api/v1beta1"
+	"gotest.tools/assert"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type FakeReconciler struct {
+	Reconciled bool
+}
+
+func (r *FakeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	r.Reconciled = true
+	return ctrl.Result{}, nil
+}
+
+func (r *FakeReconciler) Reset() {
+	r.Reconciled = false
+}
+
+type SecretReconcilerTest struct {
```

Comment:
maybe no need to export this, since it's quite specific to these batch of tests.

```suggestion
type secretReconcilerTest struct {
```